src/components/searchInput/types.ts

```diff
@@ -0,0 +1,72 @@
+import {TextInputProps, StyleProp, ViewStyle, TextStyle, ActivityIndicatorProps} from 'react-native';
+import {ButtonProps} from '../button';
+
+export enum SearchInputPresets {
+  DEFAULT = 'default',
+  PROMINENT = 'prominent'
+}
+
+export interface SearchInputRef {
+  blur: () => void;
+  clear: () => void;
+  focus: () => void;
+}
+
+export type SearchInputProps = TextInputProps & {
+  /**
+   * On clear button callback.
+   */
+  onClear?: () => void;
+  /**
+   * callback for dismiss action
+   */
+  onDismiss?: () => void;
+  /**
+   * Props for the cancel button
+   */
+  cancelButtonProps?: ButtonProps;
+  /**
+   * Custom right element
+   */
+  renderCustomRightElement?: (props?: any) => React.ReactElement;
+  /**
+   * Title prop
+   */
+  title?: string;
+  /**
+   * Whether to show a loader instead of the left search icon
+   */
+  showLoader?: boolean;
+  /**
+   * Loader props
+   */
+  loaderProps?: ActivityIndicatorProps;
+  /**
+   * custom loader element
+   */
+  renderLoaderElement?: (props?: any) => React.ReactElement;
```

Comment:
customLoader, I don't think you need to pass back the props here

Reply:
?

Reply:
You didn't change the prop here at all, not the name and not the type... why is it a function if you don't return anything?

Reply:
Changed to `customLoader: React.ReactElement`.
Function is not relevant here.